src/components/DoctorCardList/DoctorCardList.tsx

```diff
@@ -6,22 +6,76 @@ import {DoctorModel} from '../../models/doctor.model';
 const DoctorCardList = () => {
     const [doctors, setDoctors] = useState<DoctorModel[]>([]);
     const [loading, setLoading] = useState(true);
+    const [searchQuery, setSearchQuery] = useState('');
+    const [filteredDoctors, setFilteredDoctors] = useState<DoctorModel[]>([]);
 
     useEffect(() => {
         setDoctors(dummyDoctorsList);
+        setFilteredDoctors(dummyDoctorsList);
         setLoading(false);
     }, []);
 
+    const handleSearchChange = (event: React.ChangeEvent<HTMLInputElement>) => {
```

Comment:
could you add debounce for search, also you can use hooks UceCallback or UseMemo for optimization